internal/jobs/noop_job.go

```diff
@@ -14,28 +14,36 @@ type NoopJobArgs struct {
 	ReservationID int64 `json:"reservation_id"`
 }
 
+// Unmarshall arguments and handle error
 func HandleNoop(ctx context.Context, job dejq.Job) error {
-	ctxLogger := ctxval.Logger(ctx)
-	ctxLogger.Debug().Msg("Started no operation job")
-
 	args := NoopJobArgs{}
 	err := job.Decode(&args)
 	if err != nil {
-		ctxLogger.Error().Err(err).Msg("unable to decode arguments")
+		ctxval.Logger(ctx).Error().Err(err).Msg("unable to decode arguments")
 		return fmt.Errorf("unable to decode args: %w", err)
 	}
-	logger := ctxLogger.With().Int64("reservation", args.ReservationID).Logger()
+	logger := ctxval.Logger(ctx).With().Int64("reservation_id", args.ReservationID).Logger()
 	logger.Info().Interface("args", args).Msg("Processing no operation job")
+	ctx = ctxval.WithAccountId(ctx, args.AccountID)
 
-	// do nothing and update status
-	rDao, err := dao.GetReservationDao(ctx)
-	if err != nil {
-		return fmt.Errorf("cannot get reservation DAO: %w", err)
-	}
-	err = rDao.Finish(ctx, args.ReservationID, true, "Finished")
-	if err != nil {
-		return fmt.Errorf("cannot update reservation status: %w", err)
+	jobErr := handleNoop(ctx, &args)
+
+	if jobErr != nil {
+		finishWithError(ctx, args.ReservationID, jobErr)
+	} else {
+		finishWithSuccess(ctx, args.ReservationID)
 	}
+	return jobErr
+}
+
+// Job logic, when error is returned the job status is updated accordingly
+func handleNoop(ctx context.Context, args *NoopJobArgs) error {
+	// status updates before and after the code logic
```

Comment:
Do we want to add the log msgs here as we do for the launch instance and upload pubkeys?

Reply:
Again, the common function `contextLogger` now logs the following:

```go
	logger := ctxval.Logger(newContext).With().Int64("reservation_id", reservationId).Logger()
	logger.Info().Interface("args", args).Msgf("Processing job: '%s'", jobName)
```

All jobs should call it, so they will all contain this line:

    reservation_id=12345 Processing job: 'job_name'